src/components/internals/root-ref/root-ref.spec.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { render } from '../../../test-utils';
+import RootRef from './root-ref';
+
+class TestComponent extends React.Component {
+  state = {
+    show: this.props.show,
```

Comment:
Nitpick: even though internal we should follow our own conventions maybe and name is `isOpen` or `shouldShow`. I don't mind too much though.